### PR TITLE
Replace contentlayer with mdx across docs and configs

### PR DESCRIPTION
**Pull request by hirotomoyamada**
## Description

Replace contentlayer with mdx across docs and configs.

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Review by taroj1205 — COMMENTED** [review]
Most pages does not seem to show, and title is missing for http://localhost:3000/getting-started/basics
image

image

**Comment by taroj1205** [comment]
Issue above may be related to the error below.
```
Warning: Found 327 problems in 935 documents.

 └── 327 documents contain field data which didn't match the structure defined in the document type definition. (Skipping documents)

     • "community/contributing.ja.mdx" of type "Community" has the following incompatible fields:
       • order: "2\r"
     • "community/contributing.mdx" of type "Community" has the following incompatible fields:
       • order: "2\r"
     • "changelog/v0.1.1.mdx" of type "Changelog" has the following incompatible fields:
       • order: "165\r"
     • "changelog/v0.1.2.mdx" of type "Changelog" has the following incompatible fields:
       • order: "164\r"
     • "changelog/v0.2.0.mdx" of type "Changelog" has the following incompatible fields:
```

File: docs/scripts/mdx/builder.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  const map: DocumentMap = {}
+  const paths: string[] = []
+
+  await Promise.all(
+    filePaths.map(async (filePath) => {
+      const locale = getLocale(filePath)
+      const slug = getSlug(filePath)
+
+      const mdx = await generateMDX(filePath)
+
```
Comment (taroj1205) [inline docs/scripts/mdx/builder.ts:24]: We should add `.replace(/\\/g, "/")` to this to fix issue on Windows.

```suggestion

      const mdx = await generateMDX(filePath.replace(/\\/g, "/"))

```

File: docs/scripts/mdx/watcher.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+export async function watchMDX(eventName: EventName, filePath: string) {
+  if (eventName === "addDir" || eventName === "unlinkDir") return
+
+  const locale = getLocale(filePath)
+  const slug = getSlug(filePath)
+
+  if (eventName === "add" || eventName === "change") {
+    const mdx = await generateMDX(filePath)
+
```
Comment (taroj1205) [inline docs/scripts/mdx/watcher.ts:18]: We should replace backslashes with `/` for Windows.

```suggestion
  if (eventName === "add" || eventName === "change") {
    const mdx = await generateMDX(filePath.replace(/\\/g, "/"))

```

File: docs/scripts/mdx/utils.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+export function isDefaultLocale(locale: string) {
+  return locale === CONSTANT.I18N.DEFAULT_LOCALE
+}
+
+export function getSlug(filePath: string): string {
+  return filePath
+    .replace(/^contents\//, "")
+    .replace(new RegExp(`\(/index\)?\(.${OTHER_LOCALES}\)?.mdx$`), "")
+}
```
Comment (taroj1205) [inline docs/scripts/mdx/utils.ts:28]: It should replace backslash with `/` on Windows.

```suggestion
export function getSlug(filePath: string): string {
  return filePath
    .replace(/\\/g, "/")
    .replace(/^contents\//, "")
    .replace(new RegExp(`\(/index\)?\(.${OTHER_LOCALES}\)?.mdx$`), "")
}
```

**Review by taroj1205 — COMMENTED** [review]
Thanks for the PR! I have made a few suggestions which fixes issues on Windows, please check.

File: docs/pages/examples/dashboard/header.tsx [hunk 4]
```diff
@@ -1,3 +1,4 @@
+import { Menu as MenuIcon } from "@yamada-ui/lucide"
```
[the remark this answers is not in the document]
Comment (hirotomoyamada) [reply docs/pages/examples/dashboard/header.tsx:1]: Fixed

File: docs/pages/icons/list.tsx [hunk 5]
```diff
@@ -1,3 +1,6 @@
+import type { IconNames } from "@yamada-ui/lucide"
+import { icons, Search as SearchIcon } from "@yamada-ui/lucide"
```
[the remark this answers is not in the document]
Comment (hirotomoyamada) [reply docs/pages/icons/list.tsx:2]: Fixed

File: docs/scripts/mdx/builder.ts (hunk 1 above)
Comment (hirotomoyamada) [reply docs/scripts/mdx/builder.ts:24]: Fixed

File: docs/scripts/mdx/watcher.ts (hunk 2 above)
Comment (hirotomoyamada) [reply docs/scripts/mdx/watcher.ts:18]: Fixed

File: docs/scripts/mdx/document.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    is_active: is_active ?? true,
+    with_table_of_contents: with_table_of_contents ?? true,
+    with_children_description: with_children_description ?? true,
+    order: order ?? 530000,
+    body: content,
+    paths,
+    slug: "/" + slug,
+    is_expanded: false,
+    edit_url: `${CONSTANT.SNS.GITHUB.DOC_EDIT_URL}/${filePath}`,
+    contents: getTableOfContents(content, table_of_contents_max_lv),
```
Comment (taroj1205) [inline docs/scripts/mdx/document.ts:36]: We should remove contents from the filePath used in this `edit_url`.

```suggestion
    is_expanded: false,
    edit_url: `${CONSTANT.SNS.GITHUB.DOC_EDIT_URL}/${filePath.replace(/^contents\//, "")}`,
    contents: getTableOfContents(content, table_of_contents_max_lv),
```

Comment (hirotomoyamada) [reply docs/scripts/mdx/document.ts:36]: Fixed

Comment (hirotomoyamada) [reply docs/scripts/mdx/document.ts:38]: Fixed

**Review by taroj1205 — APPROVED** [review]
LGTM Thanks!

**Pull request merged**